09_json/willshen8/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	jsonify "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/json"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "Puppy Json File Name").Short('d').String()
+)
+
+func main() {
+	processCommandLineArgs()
+}
+
+func processCommandLineArgs() {
```

Comment:
I'd make this have an argument `args []string` for easier unit testing.
I'd also make it return a `Config` struct with all relevant configuration information parsed from args. in main I'd then call the actual store creation etc, maybe wrap it all a `func CreatePuppies(store Storer, reader io.Reader) error` or so.

Reply:
Thanks, I agree that having a big function that does everything is not ideal. Hence I have broken it down into smaller functions that do more specific jobs. As we only returning the filename as configuration information, I won't use it in lab09, but definitely use it in lab10. Thanks.